src/NUnitFramework/framework/Internal/Execution/CompositeWorkItem.cs

```diff
@@ -231,13 +236,32 @@ private void RunChildren()
 
         private void CreateChildWorkItems()
         {
-            _children = new List<WorkItem>();
+            Children = new List<WorkItem>();
 
             foreach (ITest test in _suite.Tests)
                 if (_childFilter.Pass(test))
-                    _children.Add(WorkItem.CreateWorkItem(test, _childFilter));
+                {
+                    Children.Add(WorkItem.CreateWorkItem(test, _childFilter));
+                    if (test.Properties.ContainsKey(PropertyNames.Order)) CountOrder++;
+                }
+
```

Comment:
Did you try my original suggestion of Inserting the ordered items at the start and Adding the non-ordered ones at the end? Once you do that, you only have to sort the first n items in the list, all of which have an Order property. This is supported by List.Sort and would leave the remaining items untouched. Additionally, ChildrenCount would then probably just be a local variable, which makes more sense since it's not used anywhere else.